body: panic when mixing data types when building a body

The following example triggers a panic as it mixes data types (e.g. number with an object).

```sh
c8y devices create --data one=1 --data one.two=null --dry
```

**Output**

```sh
panic: interface conversion: interface {} is float64, not map[string]interface {}
goroutine 1 [running]:
github.com/reubenmiller/go-c8y-cli/v2/pkg/jsonUtilities.setValue(0x14000501860, {0x140005b209a?, 0x103372a50?}, {0x0?, 0x0})
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/jsonUtilities/shorthand.go:216 +0x1e8
github.com/reubenmiller/go-c8y-cli/v2/pkg/jsonUtilities.parseShorthandJSONStructure({0x140005b209a, 0xc}, 0x140005b209a?)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/jsonUtilities/shorthand.go:190 +0x16c
github.com/reubenmiller/go-c8y-cli/v2/pkg/jsonUtilities.ParseJSON({0x140005b209a, 0xc}, 0x14000501860)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/jsonUtilities/shorthand.go:57 +0x8c
github.com/reubenmiller/go-c8y-cli/v2/pkg/flags.WithDataValueAdvanced.func1(0x1400038c7e0?, 0x1031d4eb7?)
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/flags/getters.go:896 +0x2b8
github.com/reubenmiller/go-c8y-cli/v2/pkg/flags.WithBody(0x0?, 0x1400038c7e0, 0x14000501140, {0x140000b5798, 0x8, 0x102b38934?})
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/flags/getters.go:148 +0xb0
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/devices/create.(*CreateCmd).RunE(0x1400071e680, 0x0?, {0x0?, 0x0?, 0x0?})
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/cmd/devices/create/create.auto.go:137 +0x768
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd/subcommand.NewSubCommand.func1(0x14000730f00?, {0x140003755f0, 0x0, 0x9})
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/cmd/subcommand/subcommand.go:55 +0x6c
github.com/spf13/cobra.(*Command).execute(0x14000730f00, {0x14000375560, 0x9, 0x9})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:916 +0x5f4
github.com/spf13/cobra.(*Command).ExecuteC(0x140003a4000)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:1040 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:968
github.com/reubenmiller/go-c8y-cli/v2/pkg/cmd.MainRun()
	/home/runner/work/go-c8y-cli/go-c8y-cli/pkg/cmd/cmd.go:101 +0x188
main.main()
	/home/runner/work/go-c8y-cli/go-c8y-cli/cmd/c8y/main.go:8 +0x20
```

--- a/pkg/jsonUtilities/shorthand.go
+++ b/pkg/jsonUtilities/shorthand.go
@@ -187,7 +187,9 @@ func parseShorthandJSONStructure(value string, data map[string]interface{}) erro
 
 	for i := 0; i < len(outputValues); i += 2 {
 		key := strings.Trim(outputValues[i], " ")
-		setValue(data, key, parseValue(outputValues[i+1]))
+		if setErr := setValue(data, key, parseValue(outputValues[i+1])); setErr != nil {
+			return setErr
+		}
 		// data[key] = parseValue(outputValues[i+1])
 		validItems++
 	}
@@ -201,7 +203,7 @@ func parseShorthandJSONStructure(value string, data map[string]interface{}) erro
 	return nil
 }
 
-func setValue(data map[string]interface{}, path string, value interface{}) {
+func setValue(data map[string]interface{}, path string, value interface{}) error {
 	keys := strings.Split(path, Separator)
 	currentMap := data
 
@@ -213,10 +215,22 @@ func setValue(data map[string]interface{}, path string, value interface{}) {
 				if _, ok := currentMap[key]; !ok {
 					currentMap[key] = make(map[string]interface{})
 				}
-				currentMap = currentMap[key].(map[string]interface{})
+				// check if type is as exampled
+				if cm, ok := currentMap[key].(map[string]interface{}); ok {
+					currentMap = cm
+				} else {
+					// throw an error if users are trying to write an object
+					// to an existing non-object field (e.g. string/float etc.)
+					return fmt.Errorf(
+						"mixed types detected. trying to assign an object to a %T. path=%s",
+						currentMap[key],
+						strings.Join(keys[0:i+1], Separator),
+					)
+				}
 			} else {
 				currentMap[key] = value
 			}
 		}
 	}
+	return nil
 }

--- a/tests/manual/common/flag_data.yaml
+++ b/tests/manual/common/flag_data.yaml
@@ -64,3 +64,11 @@ tests:
             "subtypes": ["linuxA", "linuxB"]
           }
         }
+
+  It should not panic when mixing data types:
+    command: |
+      c8y devices create -n --data one=1 --data one.two=null --dry
+    exit-code: 101
+    stderr:
+      not-contains:
+        - panic

